src/CaptureServiceBase/CloudCollectorStartStopCaptureRequestWaiter.cpp

```diff
@@ -13,6 +13,7 @@ namespace orbit_capture_service_base {
 CaptureOptions CloudCollectorStartStopCaptureRequestWaiter::WaitForStartCaptureRequest() {
   absl::MutexLock lock(&start_mutex_);
   start_mutex_.Await(absl::Condition(&start_requested_));
+  capture_start_time_ = absl::Now();
```

Comment:
A possible simplification is to just use `AwaitWithTimeout` in `WaitForStopCaptureRequest` below and get rid of this field (which might also be a bit confusing as we have a similar `capture_start_unix_time_ns` in `CaptureStarted`). After all you don't need to be too precise and the only difference is between including the capture initialization time or not.

Reply:
Thanks for the suggestion. But then the `CloudCollectorStartStopCaptureRequestWaiter` needs to access the CaptureStarted event in `WaitForStopCaptureRequest`.